demos/social_distance_demo/cpp/main.cpp

```diff
@@ -214,16 +202,27 @@ class Drawer : public Task {
 // draws results on the frame
 class ResAggregator : public Task {
 public:
-    ResAggregator(const VideoFrame::Ptr &sharedVideoFrame, std::list<cv::Rect> &&boxes,
-                  std::list<TrackableObject> &&trackables)
-            : Task{sharedVideoFrame, 4.0},
-              boxes{std::move(boxes)},
-              trackables{std::move(trackables)} {}
+    ResAggregator(const VideoFrame::Ptr& sharedVideoFrame, std::list<cv::Rect>&& boxes,
+        std::list<TrackableObject>&& trackables)
+        : Task{ sharedVideoFrame, 4.0 },
+        context(static_cast<ReborningVideoFrame*>(sharedVideoFrame.get())->context),
+        sourceID(sharedVideoFrame->sourceID),
+        frameID(sharedVideoFrame->frameId),
+        lastProcessedFrameID(context.trackersContext.lastProcessedIDs[sourceID]),
+        boxes{std::move(boxes)},
+        trackables{std::move(trackables)} {}
+
+    bool isReady() override {
+        return lastProcessedFrameID == frameID;
+    }
 
-    bool isReady() override { return true; }
     void process() override;
 
 private:
+    Context& context;
+    const unsigned sourceID;
+    const int64_t& frameID;
```

Comment:
`ResAggregator` is a `Task`. `Task`s keep a ptr to `VideoFrame`. `VideoFrame`s have `sourceID` and `frameID`. No need to restore them as `ResAggregator`'s members.
Same for `context` and `ReborningVideoFrame`.

Reply:
Ok, removed.